service/src/main/java/bio/terra/workspace/service/resource/model/WsmResourceHandler.java

```diff
@@ -17,4 +19,13 @@ public interface WsmResourceHandler {
    * @return resource object
    */
   WsmResource makeResourceFromDb(DbResource dbResource);
+
+  /**
+   * Generate the resource cloud-native name for resource
+   *
+   * @param workspaceUuid workspace UUID
```

Comment:
nit, can you add a sentence here: when not null, the generated name will attach workspace project id.

Reply:
This is the super class interface. only in bucket case. while the resource name is null. will generate resource name attach with workspace project id. 

I under stand your meaning, could we change the sentence to : when it is null, the generated name will follow the generation rules.